### PR TITLE
[GD-2020] Create ci pipeline for givesource

### PR DESCRIPTION
**Pull request by dylanmmatthews**

### CODE REVIEW

File: cloudformation/cicd-resources.yml [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+            Resource:
+              - '*'
+
+  keyAlias:
+    Type: AWS::KMS::Alias
+    Properties:
+      AliasName: alias/firespring/cicd/default
+      TargetKeyId: !GetAtt key.Arn
+
+  gihubOauthTokenParameter:
```
Comment (ebmeierj) [inline cloudformation/cicd-resources.yml:54]: It appears these parameters can all be removed.

File: cloudformation/cicd-resources.yml [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+              - StorageClass: STANDARD_IA
+                TransitionInDays: 30
+      LoggingConfiguration:
+        DestinationBucketName: !ImportValue AccessLoggingBucket
+        LogFilePrefix: !Sub cicd-artifacts-bucket-${AWS::AccountId}-${AWS::Region}
+      VersioningConfiguration:
+        Status: Enabled
+
+  # TODO: Encryption at rest - what permissions does lambda need?
+  dlqTopic:
```
Comment (ebmeierj) [inline cloudformation/cicd-resources.yml:114]: Is this used anywhere?

File: cloudformation/cicd-resources.yml [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+                  "logs:PutLogEvents"
+                ],
+                "Effect": "Allow",
+                "Principal": {
+                  "Service": [
+                    "events.amazonaws.com",
+                    "delivery.logs.amazonaws.com"
+                  ]
+                },
+                "Resource": "arn:aws:logs:region:account:log-group:/aws/events/*:*",
```
Comment (ebmeierj) [inline cloudformation/cicd-resources.yml:154]: I believe region/account need to use the aws variables. I'm assuming this is wrong in the script you copied this from as well.

File: ops/aws/cloudformation/ci/branch.yml [hunk 4]
```diff
@@ -0,0 +1,759 @@
...
+
+  BaseRepoRegion:
+    Type: AWS::SSM::Parameter
+    Properties:
+      Description: The region the ecr registry is located in
+      Name: !Sub /Firespring/CiCd/Givesource/${AWS::StackName}/BaseRepoRegion
+      Type: String
+      Value: !Ref AWS::Region
+
+  SeedRegistryId:
```
Comment (ebmeierj) [inline ops/aws/cloudformation/ci/branch.yml:53]: I don't believe we use anything from seed in givesource?

File: cloudformation/cicd-resources.yml (hunk 3 above)
Comment (dylanmmatthews) [reply cloudformation/cicd-resources.yml:154]: The [fdp-devops cicd-resources](https://github.com/firespring/fdp-devops/blob/e7176d7977085771156ba48fa4ac037a9084a852/aws/cloudformation/fdp/cicd-resources.yml#L625) has this too, along with the engineering-devops also has this.

**Review by mkeiser82 — APPROVED** [review]

**Pull request merged**